### ISSUE TITLE
[BUG] streamlit-analytics cannot track multi-page separately

### ISSUE THREAD

**Issue by Joyce920hope** [444B/streamlit-analytics2#47]
I have 2 pages in my application, and I'm tracking both the page views and st.sidebar.file_uploader for each page. On page1, the number of page_view and the number of uploaded_file is only for page 1, which is as expected. However for page2, the number of page_view and the number of uploaded_file are not only of page2, but also combined with these two metrics of page1. Any idea why that maybe happening? Thank you.

**Comment by 444B** [comment]
@Joyce920hope Thank you for raising this issue!
I believe this is is within the current limitations of streamlit_analytics2

Can you try add a tracking to the other pages and let me know if that works?
I think it might since it will still increment session state

**Comment by 444B** [comment]
Oh hang on, I understand that you want to see the metrics for each page separately... Yes, that would be happening since both calls to streamlit_analytics on each page is going to contribute to the session state as a whole.

**Comment by Joyce920hope** [comment]
@444B Yeah, I believe this is the limitation of the package. Thanks for your explanation. And thank you for taking the action! If you got any update on the issue, please let me know!

**Comment by 444B** [comment]
Related issue #100

**Comment by alon-sht** [comment]
Do you belive this issue will be solved sometimes soon?
This would be awesome
I tried playing with it by calling reset_counts() within strart_tracking()
This indeed sepeareted the info from the pages, but it resets the counts every refresh...
